Mathlib/Analysis/SpecialFunctions/Gamma/Beta.lean

```diff
@@ -458,10 +458,95 @@ theorem Gamma_ne_zero_of_re_pos {s : ℂ} (hs : 0 < re s) : Gamma s ≠ 0 := by
   contrapose! hs
   simpa only [hs, neg_re, ← ofReal_natCast, ofReal_re, neg_nonpos] using Nat.cast_nonneg _
 
+lemma integral_re {X : Type*} [MeasurableSpace X]
+    (μ : Measure X) {f : X → ℂ} (hf : Integrable f μ) :
+    ∫ x, (f x).re ∂μ = (∫x, f x ∂μ).re := _root_.integral_re hf
+
+lemma setIntegral_re {X : Type*} [MeasurableSpace X]
+    (μ : Measure X) {s : Set X} {f : X → ℂ} (hf : IntegrableOn f s μ) :
+    ∫ x in s, (f x).re ∂μ = (∫x in s, f x ∂μ).re := Complex.integral_re _ hf.integrable
+
 end Complex
 
 namespace Real
 
+/-- The `Γ` function (of a real variable `s`). -/
+@[pp_nodot]
+def Beta (a b : ℝ) : ℝ :=
+  (Complex.betaIntegral a b).re
+
+lemma integrable_iff_ofReal {X : Type*} [MeasurableSpace X] {μ : Measure X}
+    {f : X → ℝ} : Integrable f μ ↔ Integrable (fun x ↦ (f x : ℂ)) μ :=
+    ⟨fun hf ↦ hf.ofReal, fun hf ↦ hf.re⟩
+
+lemma integrableOn_iff_ofReal {X : Type*} [MeasurableSpace X] {μ : Measure X} {s : Set X}
+    {f : X → ℝ} : IntegrableOn f s μ ↔ IntegrableOn (fun x ↦ (f x : ℂ)) s μ := integrable_iff_ofReal
+
+lemma BetaIntegral_ofReal_interval {a b : ℝ} (ha : 0 < a) (hb : 0 < b) :
+    (∫ (x : ℝ) in (0)..1, (x : ℂ) ^ (↑a - 1:ℂ) * (1 - ↑x) ^ (↑b - 1:ℂ)).re =
+      ∫ (x : ℝ) in (0)..1, x ^ (a - 1) * (1 - x) ^ (b - 1) := by
```

Comment:
The reason I had this separated was since it was a fairly large chunk of code repeated btw the following lemma and Beta_pos (which I thought was a nice fact to add in the special functions). I will put this in that PR as opposed to this one

Reply:
But in `Beta_pos` you can directly `rw` with `BetaIntegral_ofReal`.

Reply:
If you have a result that's important in the context of a proof but artificial or useless elsewhere, usually the solution is to make it a `private` lemma or theorem.

Reply:
I don't think this is the way to go here, as this lemma is the main part of the proof `BetaIntegral_ofReal` and is not used elsewhere (as I said, it is not necessary for `Beta_pos`), so I think it definitely makes more sense to put it directly in the proof of `BetaIntegral_ofReal`.